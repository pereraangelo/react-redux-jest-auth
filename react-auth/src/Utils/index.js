import moment from "moment";

export const getGreetingTime = () => {
  let currentTime = new Date();

  const splitAfternoon = 12; // 24hr time to split the afternoon
  const splitEvening = 15; // 24hr time to split the evening
  const currentHour = moment(currentTime).format("HH");

  if (currentHour >= splitAfternoon && currentHour <= splitEvening) {
    // Between 12 PM and 5PM
    return "Good Afternoon";
  } else if (currentHour >= splitEvening) {
    // Between 5PM and Midnight
    return "Good Evening";
  }
  // Between Dawn and noon
  return "Good Morning";
};



export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  let input = findByTestAtrr(wrapper, inputSelector);
  input.simulate("change", {
    target: { name: inputSelector, value: newValue },
  });

  return findByTestAtrr(wrapper, inputSelector);
};


export const simulateCheckBox = (wrapper, inputSelector, newValue) => {
    let input = findByTestAtrr(wrapper, inputSelector);
    input.simulate('change', {target: {name:inputSelector,checked: newValue}});
  
    return findByTestAtrr(wrapper, inputSelector);
  };

  