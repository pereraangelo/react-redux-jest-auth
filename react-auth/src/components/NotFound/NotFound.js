import React from "react";
import { Link } from "react-router-dom";

function NotFound(props) {
  return (
    <div className="notFoundPage">
      <h1>404</h1> <p>oopz.The page you are looking for doesnot exsists.</p>
      <div>Go back to   <Link to="/" className=" btn-link m-3" data-test="login">
          {" "}
          Login{" "}
        </Link> page</div>
    </div>
  );
}

export default NotFound;
