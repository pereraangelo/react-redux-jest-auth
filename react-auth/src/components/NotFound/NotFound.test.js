import React from "react";
import ReactDOM from 'react-dom'
import {shallow} from "enzyme"; 
import {MemoryRouter} from "react-router";
import { findByTestAtrr } from "../../Utils";
import renderer from 'react-test-renderer'
import NotFound from "./NotFound";




describe('Notfound component',()=>{ 
    const wrapper = shallow( <NotFound />);
    it('Notfound commponent matches the snapshot',()=>{
        const tree=renderer.create(<MemoryRouter ><NotFound/></MemoryRouter>).toJSON()
        expect(tree).toMatchSnapshot()
    });
      
 
    it('Notfound page should load without crahing',()=>{
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter ><NotFound/></MemoryRouter>, div);
    });
     

    it('redirect user to Login page',()=>{
        const cancle = findByTestAtrr(wrapper, 'login').props().to
        
        expect(cancle).toEqual("/") 
         
    })

})