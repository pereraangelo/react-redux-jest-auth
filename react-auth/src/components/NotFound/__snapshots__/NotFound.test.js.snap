// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notfound component Notfound commponent matches the snapshot 1`] = `
<div
  className="notFoundPage"
>
  <h1>
    404
  </h1>
   
  <p>
    oopz.The page you are looking for doesnot exsists.
  </p>
  <div>
    Go back to   
    <a
      className=" btn-link m-3"
      data-test="login"
      href="/"
      onClick={[Function]}
    >
       
      Login
       
    </a>
     page
  </div>
</div>
`;
